sklearn/naive_bayes.py

```diff
@@ -351,9 +351,10 @@ def _partial_fit(self, X, y, classes=None, _refit=False,
         unique_y_in_classes = in1d(unique_y, classes)
 
         if not np.all(unique_y_in_classes):
+            not_unique_y_in_classes = ~unique_y_in_classes
             raise ValueError("The target label(s) %s in y do not exist in the "
                              "initial classes %s" %
-                             (y[~unique_y_in_classes], classes))
+                             (y[np.where(not_unique_y_in_classes)], classes))
```

Comment:
From numpy>1.8.0, boolean indices would be treated as masks and raise a FutureWarning warning. That's why i used np.where. 
Regarding regression test, could you please guide me on what is to be done ? I'm new to sklearn and am really interested in working on this. :)